utils.py

```diff
@@ -52,25 +52,48 @@ def get_scenario_params_list(config):
 
     scenario_params_list = []
     
+    # Separate scenarios from different dataset
+    config_dataset = []
     for list_scenario in config:
+        
+        if isinstance(list_scenario['dataset_name'], dict):
+            for dataset_name in list_scenario['dataset_name'].keys():
+                # Add path to init model from an existing model
+                dataset_scenario = list_scenario.copy()
+                dataset_scenario['dataset_name'] = [dataset_name]
+                dataset_scenario['init_model_from'] = list_scenario['dataset_name'][dataset_name]
+                
+                config_dataset.append(dataset_scenario)
+        else:
+            config_dataset.append(list_scenario)
```

Comment:
So the purpose of this is to sort all `list_scenarios` per dataset, before creating the `scenarios` themselves and checking the possible exceptions. I don't see where this sorting per dataset is used, why it is needed?

Reply:
I did that to correctly used the cartesian product and to avoid mixing models between datasets.